storage/src/mmr/iterator.rs

```diff
@@ -144,6 +144,37 @@ pub(crate) fn leaf_pos_to_num(leaf_pos: u64) -> u64 {
     leaf_num_floor
 }
 
+/// Returns the position of the leaf with number `leaf_num` in an MMR.
+///
+/// This computation is O(log2(n)) in `leaf_num`.
+#[allow(dead_code)] // TODO: remove this when we start using it
+pub(crate) fn leaf_num_to_pos(leaf_num: u64) -> u64 {
+    if leaf_num == 0 {
+        return 0;
+    }
+
+    // The following won't underflow because any sane leaf number would have several leading zeros.
```

Comment:
nit: In a different PR/issue, it may be worth documenting some of these "max capacity" edge cases somewhere. AFAICT we generally assume a user will never get close to `u64::MAX` (and overflow/underflow) if you do.